platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/manager/internal/StandardThreadManager.java

```diff
@@ -0,0 +1,249 @@
+import com.swirlds.common.threading.locks.AutoClosableLock;
+import com.swirlds.common.threading.locks.Locks;
+import com.swirlds.common.threading.locks.locked.Locked;
+import com.swirlds.common.threading.manager.StartableThreadManager;
+import com.swirlds.common.threading.manager.ThreadBuilder;
+import com.swirlds.common.threading.manager.ThreadManager;
+import com.swirlds.common.utility.LifecyclePhase;
+import com.swirlds.common.utility.Startable;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * A standard implementation of a {@link ThreadManager}. Will not permit work to be done on threads until started.
+ */
+public final class StandardThreadManager extends AbstractThreadManager
+        implements StartableThreadManager, ThreadBuilder {
+
+    private LifecyclePhase phase = LifecyclePhase.NOT_STARTED;
+    private final List<Startable> thingsToBeStarted = new ArrayList<>();
+
+    private final Runnable throwIfInWrongPhase = () -> this.throwIfNotInPhase(LifecyclePhase.STARTED);
+
+    private final AutoClosableLock lock = Locks.createAutoLock();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public @NonNull LifecyclePhase getLifecyclePhase() {
+        return phase;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() {
+        try (final Locked l = lock.lock()) {
+            throwIfNotInPhase(LifecyclePhase.NOT_STARTED);
+            phase = LifecyclePhase.STARTED;
+            thingsToBeStarted.forEach(Startable::start);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        try (final Locked l = lock.lock()) {
+            throwIfNotInPhase(LifecyclePhase.STARTED);
+            phase = LifecyclePhase.STOPPED;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public @NonNull Thread buildThread(@NonNull final Runnable runnable) {
+        try (final Locked l = lock.lock()) {
+            throwIfAfterPhase(LifecyclePhase.STARTED);
+            return new ManagedThread(runnable, throwIfInWrongPhase);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public ExecutorService createCachedThreadPool(
+            @NonNull final String name, @NonNull final UncaughtExceptionHandler uncaughtExceptionHandler) {
+        Objects.requireNonNull(name);
+        Objects.requireNonNull(uncaughtExceptionHandler);
+        try (final Locked l = lock.lock()) {
+            throwIfAfterPhase(LifecyclePhase.STARTED);
+            final ManagedExecutorService service = new ManagedExecutorService(
+                    Executors.newCachedThreadPool(buildThreadFactory(name, uncaughtExceptionHandler)));
```

Comment:
Similar to the Ad Hoc mirror, it would be good to avoid the convenience methods of `Executors` and create `ExecutorService` objects that have bounded queues, so we don't have to accept the risks associated with an unbounded queue.